src/libtrx/include/libtrx/game/input/state.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <stdint.h>
+
+typedef union INPUT_STATE {
+    uint64_t any;
+    struct {
+        uint64_t forward : 1;
+        uint64_t back : 1;
+        uint64_t left : 1;
+        uint64_t right : 1;
+        uint64_t jump : 1;
+        uint64_t draw : 1;
+        uint64_t action : 1;
+        uint64_t slow : 1;
+        uint64_t look : 1;
+        uint64_t step_left : 1;
+        uint64_t step_right : 1;
+        uint64_t roll : 1;
+
+        uint64_t pause : 1;
+        uint64_t enter_console : 1;
+        uint64_t save : 1;
+        uint64_t load : 1;
+
+#if TR_VERSION == 1
+        uint64_t fly_cheat : 1;
+        uint64_t item_cheat : 1;
+        uint64_t level_skip_cheat : 1;
+        uint64_t turbo_cheat : 1;
+        uint64_t camera_up : 1;
+        uint64_t camera_down : 1;
+        uint64_t camera_forward : 1;
+        uint64_t camera_back : 1;
+        uint64_t camera_left : 1;
+        uint64_t camera_right : 1;
+        uint64_t camera_reset : 1;
+        uint64_t change_target : 1;
+#endif
+
+        uint64_t equip_pistols : 1;
+        uint64_t equip_shotgun : 1;
+        uint64_t equip_magnums : 1;
+        uint64_t equip_uzis : 1;
+#if TR_VERSION == 2
+        uint64_t equip_harpoon : 1;
+        uint64_t equip_m16 : 1;
+        uint64_t equip_grenade_launcher : 1;
+#endif
+        uint64_t use_small_medi : 1;
+        uint64_t use_big_medi : 1;
+#if TR_VERSION == 2
+        uint64_t use_flare : 1;
```

Comment:
The Venice demo doesn't play out as expected: Lara doesn't draw a flare at the beginning, and once she goes through the window, her movement is wrong (she hits the wall and dies because the goon has free rein).
I am assuming this will need to remain in the original bit position (`1 << 19` I believe) otherwise the demo data in the level would need to be updated. Whether this affects her movement too, I'm not sure - maybe because it's not registering, other inputs are running sooner?

Reply:
The Lost Valley demo also has an issue - Lara doesn't exit the water.
Vilcabamba, Maria Doria, and Tibet demos look OK as far as I can tell.

Reply:
@lahm86 fixed

Reply:
TR2 looks good. The Lost Valley issue remains, but I've just checked develop and the problem is there too. I'll raise that as a separate issue and work out when it was introduced.